### PR TITLE
Use Activator.CreateInstance to create generic type at runtime

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Had to play around with the types a bit but I think this should work and allow you to keep the strong typings on the properties.

I didn't actually try it at runtime, if you could do that and things look good then merge this in and update the rest of the properties to have the correct type.

(you could also use reflection to set the properties, but making a constructor seemed easiest)

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs [hunk 1]
```diff
@@ -197,13 +205,13 @@ public class DeploymentOptions
 
         public DeploymentOptionProperty<object> IgnoreTablePartitionOptions { get; set; } // DW Specific
 
-        public DeploymentOptionProperty<object> AdditionalDeploymentContributorPaths { get; set; } = new DeploymentOptionProperty<object>();
+        public DeploymentOptionProperty<object> AdditionalDeploymentContributorPaths { get; set; } = new DeploymentOptionProperty<object>(null);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs:208]: I don't know what an appropriate values for these would be. Do you even want to initialize them here vs just leaving the property null to begin with?

**Review by kisantia — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs [hunk 2]
```diff
@@ -33,7 +41,7 @@ public class DeploymentOptions
     {
         #region Properties
 
-        public DeploymentOptionProperty<object> IgnoreTableOptions { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreTableOptions { get; set; }
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs:44]: After you merge in you should change the rest of the properties to their actual types.

**Comment by Charles-Gagnon** [comment]
Ping @ssreerama - this is meant to be merged into your branch. Let me know if you have any questions otherwise you can handle doing the merge yourself when you're ready.

File: src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs (hunk 2 above)
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs:44]: In one of my previous iterations these properties are changed to bool/string type, In runtime the property Intialization with values required type casting which ends with 'if else if' statements. I'll update them now as my final iteration(hopefully :)). thanks.

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs:44]: I'm a bit confused here - what exactly are you saying is the issue? Were you able to find a way to have strong typing without the `Activator.CreateInstance` solution I have in this PR?

We shouldn't keep these types as `object` - that kind of defeats the purpose of doing generics in the first place.

**Pull request closed without merging**